Sources/ApolloCodegenTestSupport/MockFileManager.swift

```diff
@@ -28,10 +28,31 @@ public class MockFileManager: FileManager {
   private var closures: [String: Closure] = [:]
   private var closuresToBeCalled: Set<String> = []
 
+  /// If `true` then all called closures must be mocked otherwise the call will fail. When `false` any called closure
+  /// that is not mocked will fall through to `super`. As a byproduct of `false`, all mocked closures must be called otherwise
+  /// the test will fail.
+  let strict: Bool
+
+  /// Designated initializer.
+  ///
+  /// - Parameters:
+  ///  - strict: If `true` then all called closures must be mocked otherwise the call will fail. When `false` any called closure
+  ///  that is not mocked will fall through to `super`. As a byproduct of `false`, all mocked closures must be called otherwise
+  ///  the test will fail.
+  public init(strict: Bool = true) {
+    self.strict = strict
+  }
+
+  deinit {
+    if strict == false && allClosuresCalled == false {
```

Comment:
This ensures that when `strict` is `false` and calls are allowed to fall through to `super`, that intentionally mocked closures aren't skipped, i.e.: they must be called.

Reply:
At this point, we really should just consider using Cuckoo, you're building some complicated mocks that do the same thing that framework does. Not urgent to change now, but the next time you are building a complicated mock, lets just look at using a library for it.